jumpscale/sals/nettools/__init__.py

```diff
@@ -32,117 +47,167 @@ def tcp_connection_test(ipaddr: str, port: int, timeout: Optional[int]):
     """
     conn = None
     try:
-        conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        if timeout:
-            conn.settimeout(timeout)
-        try:
-            conn.connect((ipaddr, port))
-        except BaseException:
-            return False
-    finally:
-        if conn:
-            conn.close()
-    return True
+        conn = socket.create_connection((ipaddr, port), timeout)
+    except (socket.gaierror, socket.herror):
+        # raised for address-related errors
+        raise
+    except OSError:
+        # (ConnectionRefusedError, socket.timeout, OSError)
+        return False
+    else:
+        conn.close()
+        return True
 
 
-def udp_connection_test(ipaddr: str, port: int, timeout=1, message=b"PING"):
-    """tests udp connection on specified port by sending specified message
+def udp_connection_test(ipaddr: str, port: int, timeout: Optional[int] = 1, message: Optional[bytes] = b"") -> bool:
```

Comment:
I think leaving the method signature would be better instead of leaving it empty to get a result without passing the params

Reply:
you talk here about the `message` arg, I respect your opinion. and see why you say so. but the default value b'PING' doesn't make sense to me. moreover, it doesn't work when I tested it (or I was not able to make it work with this default value).
first, it looks like a Redis specific message, and - correct me if I'm wrong - Redis uses TCP, and this function to test UDP connections.

second, even if it could work some way with Redis, it still doesn't make sense to me, because it is a general function. so if you want to test a specific UDP connection, you should provide a message, that the service you try to connect to, would understand it properly.

Reply:
fair enough